tests/system/test_openai_encoder.py

```diff
@@ -0,0 +1,83 @@
+import pytest
+from pinecone_text.dense import OpenAIEncoder
+from openai import BadRequestError, AuthenticationError
+
+
+DEFAULT_DIMENSION = 1536
+
+
+@pytest.fixture
+def openai_encoder():
+    return OpenAIEncoder()
+
+
+def test_init_with_kwargs():
+    encoder = OpenAIEncoder(
+        api_key="test_api_key", organization="test_organization", timeout=30
+    )
+    assert encoder._client.api_key == "test_api_key"
+    assert encoder._client.organization == "test_organization"
+    assert encoder._client.timeout == 30
+
+    with pytest.raises(AuthenticationError):
+        encoder.encode_documents("test text")
+
+
+def encode_by_type(openai_encoder, encoding_function, test_input):
+    if encoding_function == "encode_documents":
+        return openai_encoder.encode_documents(test_input)
+    elif encoding_function == "encode_queries":
+        return openai_encoder.encode_queries(test_input)
+    pytest.fail(f"Unknown encoding function: {encoding_function}")
```

Comment:
I think you can do this:
```suggestion
    func = getattr(openai_encoder, encoding_function)
    return func(test_input)
```